.github/workflows/docker.yaml

```diff
@@ -0,0 +1,48 @@
+name: Docker
+on:
+  pull_request:
+    paths:
+      - add_me_to_your_PATH/Dockerfile.template
+      - add_me_to_your_PATH/sysimage.jl
+      - add_me_to_your_PATH/startup.jl
+      - .github/workflows/docker.yaml
+
+# Each PR and each commit on `main` use distinct concurrency groups. Note canceling workflows in
+# progress may result in having to rebuild layers as the Docker layer caching is only uploaded
+# once the image has been fully built. That said, new commits may invalidate these layers anyway.
+concurrency:
+  group: image-build-${{ github.ref_name }}-${{ github.ref == 'refs/heads/main' && github.run_number }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: Image Build
+    runs-on: ubuntu-latest
+    env:
+      # Reference the HEAD commit which triggerred this workflow. By default PRs use a merge commit
+      SHA: ${{ github.event.pull_request.head.sha || github.sha }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ env.SHA }}  # Always checkout HEAD commit
```

Comment:
In this case we could and it wouldn't make a difference. For some Docker builds though you may include the `.git` folder so you can determine the SHA used within the image. In these scenarios using the merge commit would result in the SHA in the image being incorrect. As it seems that the `julia_pod` code is somewhat of a "jumping off point" adding this was mostly defensive to avoid confusion from users copying this workflow.